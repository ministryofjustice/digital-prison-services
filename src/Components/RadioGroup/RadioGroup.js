// Probably a good candidate for new-nomis-shared if used elsewhere
import React from 'react'
import PropTypes from 'prop-types'

import MultiChoice from '@govuk-react/multi-choice'
import Radio from '@govuk-react/radio'

const RadioGroup = ({ label, hint, options, inline, input, meta }) => (
  <MultiChoice label={label} hint={hint} meta={meta} mb={6}>
    {options.map(o => (
      <div key={o.value}>
        <Radio {...input} value={o.value} inline={inline} checked={o.value === input.value}>
          {o.title}
        </Radio>
      </div>
    ))}
  </MultiChoice>
)

RadioGroup.defaultProps = {
  input: {},
  meta: {},
  label: undefined,
  hint: undefined,
  inline: false,
  options: [],
}

RadioGroup.propTypes = {
  input: PropTypes.shape({
    value: PropTypes.string,
  }),
  meta: PropTypes.shape({}),
  label: PropTypes.string,
  hint: PropTypes.string,
  inline: PropTypes.bool,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string,
      value: PropTypes.string,
    })
  ),
}

export default RadioGroup
