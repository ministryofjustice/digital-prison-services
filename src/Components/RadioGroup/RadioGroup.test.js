import React from 'react'
import renderer from 'react-test-renderer'
import RadioGroup from '.'

describe('<RadioGroup />', () => {
  const props = {
    label: 'Does this test pass?',
    options: [{ title: 'Yes', value: 'yes' }, { title: 'No', value: 'no' }],
  }

  it('should match the default snapshot', () => {
    const tree = renderer.create(<RadioGroup {...props} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should match the with hint text snapshot', () => {
    const tree = renderer.create(<RadioGroup {...props} hint="Supporting text to help" />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should match the with meta and error snapshot', () => {
    const tree = renderer.create(<RadioGroup {...props} meta={{ touched: true, error: 'Select an option' }} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should display image if one is provided', () => {
    const propsImage = {
      label: 'Does this test pass?',
      options: [{ title: 'Yes', value: 'yes', image: 'image.png' }, { title: 'No', value: 'no', image: 'image1.png' }],
    }

    const tree = renderer
      .create(<RadioGroup {...propsImage} meta={{ touched: true, error: 'Select an option' }} />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
