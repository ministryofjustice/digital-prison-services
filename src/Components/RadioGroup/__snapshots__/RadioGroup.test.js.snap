// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioGroup /> should display image if one is provided 1`] = `
<div
  className="sc-bxivhb dgxkrU"
>
  <span
    className="sc-bdVaJa jmnRGq"
  >
    Does this test pass?
  </span>
  <span
    className="sc-bwzfXH bXamfX"
  >
    Select an option
  </span>
  <div>
    <label
      className="sc-ifAKCX irKPnh"
    >
      <input
        checked={false}
        className="sc-EHOje koflXw"
        type="radio"
        value="yes"
      />
      <span
        className="sc-bZQynM ldmnhp"
      >
        Yes
      </span>
    </label>
  </div>
  <div>
    <label
      className="sc-ifAKCX irKPnh"
    >
      <input
        checked={false}
        className="sc-EHOje koflXw"
        type="radio"
        value="no"
      />
      <span
        className="sc-bZQynM ldmnhp"
      >
        No
      </span>
    </label>
  </div>
</div>
`;

exports[`<RadioGroup /> should match the default snapshot 1`] = `
<div
  className="sc-bxivhb hwZAGr"
>
  <span
    className="sc-bdVaJa jmnRGq"
  >
    Does this test pass?
  </span>
  <div>
    <label
      className="sc-ifAKCX irKPnh"
    >
      <input
        checked={false}
        className="sc-EHOje koflXw"
        type="radio"
        value="yes"
      />
      <span
        className="sc-bZQynM ldmnhp"
      >
        Yes
      </span>
    </label>
  </div>
  <div>
    <label
      className="sc-ifAKCX irKPnh"
    >
      <input
        checked={false}
        className="sc-EHOje koflXw"
        type="radio"
        value="no"
      />
      <span
        className="sc-bZQynM ldmnhp"
      >
        No
      </span>
    </label>
  </div>
</div>
`;

exports[`<RadioGroup /> should match the with hint text snapshot 1`] = `
<div
  className="sc-bxivhb hwZAGr"
>
  <span
    className="sc-bdVaJa jmnRGq"
  >
    Does this test pass?
  </span>
  <span
    className="sc-htpNat cFrViK"
  >
    Supporting text to help
  </span>
  <div>
    <label
      className="sc-ifAKCX irKPnh"
    >
      <input
        checked={false}
        className="sc-EHOje koflXw"
        type="radio"
        value="yes"
      />
      <span
        className="sc-bZQynM ldmnhp"
      >
        Yes
      </span>
    </label>
  </div>
  <div>
    <label
      className="sc-ifAKCX irKPnh"
    >
      <input
        checked={false}
        className="sc-EHOje koflXw"
        type="radio"
        value="no"
      />
      <span
        className="sc-bZQynM ldmnhp"
      >
        No
      </span>
    </label>
  </div>
</div>
`;

exports[`<RadioGroup /> should match the with meta and error snapshot 1`] = `
<div
  className="sc-bxivhb dgxkrU"
>
  <span
    className="sc-bdVaJa jmnRGq"
  >
    Does this test pass?
  </span>
  <span
    className="sc-bwzfXH bXamfX"
  >
    Select an option
  </span>
  <div>
    <label
      className="sc-ifAKCX irKPnh"
    >
      <input
        checked={false}
        className="sc-EHOje koflXw"
        type="radio"
        value="yes"
      />
      <span
        className="sc-bZQynM ldmnhp"
      >
        Yes
      </span>
    </label>
  </div>
  <div>
    <label
      className="sc-ifAKCX irKPnh"
    >
      <input
        checked={false}
        className="sc-EHOje koflXw"
        type="radio"
        value="no"
      />
      <span
        className="sc-bZQynM ldmnhp"
      >
        No
      </span>
    </label>
  </div>
</div>
`;
